sharding/nodesCoordinator/indexHashedNodesCoordinatorLite_test.go

```diff
@@ -0,0 +1,1217 @@
+	require.NoError(t, err)
+	require.True(t, setTypeWasCalled)
+	require.Equal(t, core.NodeTypeObserver, nodeTypeResult)
+}
+
+func TestIndexHashedNodesCoordinator_GetConsensusValidatorsPublicKeysNotExistingEpoch(t *testing.T) {
+	t.Parallel()
+
+	args := createArguments()
+	ihgs, err := NewIndexHashedNodesCoordinatorLite(args)
+	require.Nil(t, err)
+
+	var pKeys []string
+	randomness := []byte("randomness")
+	pKeys, err = ihgs.GetConsensusValidatorsPublicKeys(randomness, 0, 0, 1)
+	require.True(t, errors.Is(err, ErrEpochNodesConfigDoesNotExist))
+	require.Nil(t, pKeys)
+}
+
+func TestIndexHashedNodesCoordinator_GetConsensusValidatorsPublicKeysExistingEpoch(t *testing.T) {
+	t.Parallel()
+
+	args := createArguments()
+	ihgs, err := NewIndexHashedNodesCoordinatorLite(args)
+	require.Nil(t, err)
+
+	shard0PubKeys := validatorsPubKeys(args.EligibleNodes[0])
+
+	var pKeys []string
+	randomness := []byte("randomness")
+	pKeys, err = ihgs.GetConsensusValidatorsPublicKeys(randomness, 0, 0, 0)
+	require.Nil(t, err)
+	require.True(t, len(pKeys) > 0)
+	require.True(t, isStringSubgroup(pKeys, shard0PubKeys))
+}
+
+func TestIndexHashedNodesCoordinator_GetValidatorsIndexes(t *testing.T) {
+	t.Parallel()
+
+	args := createArguments()
+	ihgs, err := NewIndexHashedNodesCoordinatorLite(args)
+	require.Nil(t, err)
+	randomness := []byte("randomness")
+
+	var pKeys []string
+	pKeys, err = ihgs.GetConsensusValidatorsPublicKeys(randomness, 0, 0, 0)
+	require.Nil(t, err)
+
+	var indexes []uint64
+	indexes, err = ihgs.GetValidatorsIndexes(pKeys, 0)
+	require.Nil(t, err)
+	require.Equal(t, len(pKeys), len(indexes))
+}
+
+func TestIndexHashedNodesCoordinator_GetValidatorsIndexesInvalidPubKey(t *testing.T) {
+	t.Parallel()
+
+	args := createArguments()
+	ihgs, err := NewIndexHashedNodesCoordinatorLite(args)
+	require.Nil(t, err)
+	randomness := []byte("randomness")
+
+	var pKeys []string
+	pKeys, err = ihgs.GetConsensusValidatorsPublicKeys(randomness, 0, 0, 0)
+	require.Nil(t, err)
+
+	var indexes []uint64
+	pKeys[0] = "dummy"
+	indexes, err = ihgs.GetValidatorsIndexes(pKeys, 0)
+	require.Equal(t, ErrInvalidNumberPubKeys, err)
+	require.Nil(t, indexes)
+}
+
+func TestIndexHashedNodesCoordinator_ShardIdForEpochInvalidEpoch(t *testing.T) {
+	t.Parallel()
+
+	args := createArguments()
+	ihgs, err := NewIndexHashedNodesCoordinatorLite(args)
+	require.Nil(t, err)
+
+	shardId, err := ihgs.ShardIdForEpoch(1)
+	require.True(t, errors.Is(err, ErrEpochNodesConfigDoesNotExist))
+	require.Equal(t, uint32(0), shardId)
+}
+
+func TestIndexHashedNodesCoordinator_ShardIdForEpochValidEpoch(t *testing.T) {
+	t.Parallel()
+
+	args := createArguments()
+	ihgs, err := NewIndexHashedNodesCoordinatorLite(args)
+	require.Nil(t, err)
+
+	shardId, err := ihgs.ShardIdForEpoch(0)
+	require.Nil(t, err)
+	require.Equal(t, uint32(0), shardId)
+}
+
+// func TestIndexHashedNodesCoordinator_GetConsensusWhitelistedNodesEpoch0(t *testing.T) {
```

Comment:
Why commented tests?

Reply:
duplicate tests, removed the commeted lines :+1: